include/silo/preprocessing/preprocessing_database.h

```diff
@@ -11,31 +11,33 @@ namespace silo {
 
 class ZstdFastaTable;
 class ReferenceGenomes;
+class CompressSequence;
 
 namespace preprocessing {
 
 class Partitions;
 
 class PreprocessingDatabase {
   public:
-   static constexpr std::string_view COMPRESS_NUC = "compressNuc";
-   static constexpr std::string_view COMPRESS_AA = "compressAA";
+   std::unique_ptr<CompressSequence> compress_nucleotide_function;
+   std::unique_ptr<CompressSequence> compress_amino_acid_function;
```

Comment:
Does this actually need to be a `unique_ptr`? I don't think this is a blocker here or something, but I just wanted to mention it, as when we extend it to multiple functions, we will not need to store a `vector<unique_ptr>`/`map<unique_ptr>`, but can store the objects in the container directly

Reply:
As discussed: It was only a pointer, so we can extend this in the future with more of the SqlFunctions. However, we should address this in #282